aten/src/ATen/ParallelNative.h

```diff
@@ -60,7 +60,10 @@ inline void parallel_for(
       internal::_unset_thread_num();
     };
 
-    std::vector<c10::ivalue::Future> futures(num_tasks);
+    std::vector<std::shared_ptr<c10::ivalue::Future>> futures(num_tasks);
+    for (size_t i = 0; i < num_tasks; ++i) {
+      futures[i] = std::make_shared<c10::ivalue::Future>(NoneType::get());
+    }
```

Comment:
@ilia-cher let me know if I should move this to a common util function, maybe in Parallel.h? I can reuse it in ParallelNativeMobile.h as well.